plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -492,6 +505,36 @@ public void testAggregationPushdown()
         assertThat(query("SELECT approx_set(nationkey) FROM nation")).isNotFullyPushedDown(AggregationNode.class);
     }
 
+    @Test
+    public void testTopNPushdown()
+    {
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey limit 10")).ordered().isFullyPushedDown();
+        assertThat(query("SELECT orderkey FROM orders ORDER BY orderkey DESC limit 10")).ordered().isFullyPushedDown();
+
+        // multiple sort columns with different orders
+        assertThat(query("SELECT * FROM orders ORDER BY orderpriority DESC, totalprice ASC LIMIT 10")).ordered().isFullyPushedDown();
+
+        // TopN over aggregation column
+        assertThat(query("SELECT sum(totalprice) AS total FROM orders GROUP BY custkey ORDER BY total DESC LIMIT 10")).ordered().isFullyPushedDown();
+
+        // TopN over TopN
+        assertThat(query("SELECT orderkey, totalprice FROM (SELECT orderkey, totalprice FROM orders ORDER BY 1, 2 LIMIT 10) ORDER BY 2, 1 LIMIT 5")).ordered().isFullyPushedDown();
+        assertThat(query("SELECT orderkey, totalprice FROM (SELECT orderkey, totalprice FROM (SELECT orderkey, totalprice FROM orders ORDER BY 1, 2 LIMIT 10) ORDER BY 2, 1 LIMIT 5) ORDER BY 1, 2 LIMIT 3")).ordered().isFullyPushedDown();
+
+        // TopN over limit
+        assertThat(query("SELECT orderkey, totalprice FROM (SELECT orderkey, totalprice FROM orders LIMIT 20) ORDER BY totalprice ASC LIMIT 5")).ordered().isFullyPushedDown();
+
+        // TopN over limit with filter
+        assertThat(query("SELECT orderkey, totalprice FROM (SELECT orderkey, totalprice FROM orders WHERE orderpriority = '1-URGENT' LIMIT 20) ORDER BY totalprice ASC LIMIT 5")).ordered().isFullyPushedDown();
+
+        // TopN over aggregation with filter
+        assertThat(query("SELECT * FROM (SELECT SUM(totalprice) as sum, custkey AS total FROM orders GROUP BY custkey HAVING COUNT(*) > 3) ORDER BY sum DESC LIMIT 10")).ordered().isFullyPushedDown();
+
+        // Filter TopN
+        assertThat(query("SELECT * FROM (SELECT * FROM orders ORDER BY totalprice DESC LIMIT 20) WHERE orderpriority = '1-URGENT'")).ordered().isFullyPushedDown();
```

Comment:
This belongs to testPredicatePushdown -- my rule is, query goes into testXxxPushdown when Xxx is the top-level feature (last to be pushed down). This is arbitrary, but let's stick to this for now